src/ripple/basics/strHex.h

```diff
@@ -62,22 +67,31 @@ charUnHex (char c)
 }
 /** @} */
 
-// NIKB TODO cleanup this function and reduce the need for the many overloads
-//           it has in various places.
-template<class FwdIt>
-std::string strHex (FwdIt first, int size)
+template<class InputIterator>
+std::string strHex(InputIterator begin, InputIterator end)
 {
-    std::string s;
-    s.resize (size * 2);
-    for (int i = 0; i < size; i++)
-    {
-        unsigned char c = *first++;
-        s[i * 2]     = charHex (c >> 4);
-        s[i * 2 + 1] = charHex (c & 15);
-    }
-    return s;
+    std::string result{};
```

Comment:
can we still pre-size the string maybe using `std::distance` ?

Reply:
Gave it a quick try using

```
std::string result{};
result.resize(std::distance(begin, end));
boost::algorithm::hex(begin, end, result.begin());
return result;
```

but that didn't quite work. A lot of test suites crash and fail to complete. I'm not sure why that is the case yet. Any ideas?

Reply:
maybe try `reserve` instead of `resize` and still use `back_inserter`

Reply:
Yep, that worked. I often confuse myself with `reserve` vs `resize` semantics. Sorry about that.

Your suggestion of pre-sizing the string is incorporated in https://github.com/ripple/rippled/pull/2668/commits/de90a69d1e083742f9b26aaffa33a93e09806fda